src/Components/Resource/ResourceCreate.tsx

```diff
@@ -202,16 +193,18 @@ export default function ResourceCreate(props: resourceProps) {
         requested_quantity: state.form.requested_quantity || 0,
       };
 
-      const res = await dispatchAction(createResource(data));
+      const { res, data } = await request(routes.createResource, {
+        body: resourceData,
+      });
       setIsLoading(false);
 
-      if (res && res.data && (res.status == 201 || res.status == 200)) {
+      if (res && data && (res.status == 201 || res.status == 200)) {
```

Comment:
```suggestion
      if (res?.ok && data) {
```

Check for `.ok` which will cover al HTTP success codes.